juju/juju
# Ensure k8s provider works with Juju managed units

## Description of change

The k8s CAAS provider is switched to having Juju mange pod creation, rather than using a deployment controller and having k8s do all that.
The main changes are passing back to the caasunitprovisioner facade the unit tag from the pod (placed there by Juju) and using that instead of provider id to sync up the juju model with the k8s state.
As a driveby, we avoid always deleting and restarting the Juju operator upon agent restart.

## QA steps

Run up a CAAS deployment.
Add/delete units from juju, check that the service is created and the pods are created/deleted.
Delete a pod and check that this is reflected in Juju.
